src/Models/Link.php

```diff
@@ -290,4 +296,23 @@ private function getLinkTypes(): array
 
         return $types;
     }
+
+    public function getDefaultTitle(): string
+    {
+        // If we have a title or don't have link, we can just bail out without any changes
+        if (!$this->config()->get('use_default_title') || empty($this->getURL()) || $this->Title) {
+            return '';
+        }
+        
+        $defaultLinkTitle = $this->getLinkTypeDefaultTitle();
+
+        $this->extend('updateDefaultLinkTitle', $defaultLinkTitle);
+
+        return $defaultLinkTitle;
+    }
+
+    public function getLinkTypeDefaultTitle(): string
+    {
+        return $this->config()->get('default_link_title');
```

Comment:
```suggestion
        $default = $this->getDescription() ?: $this->getURL();
        if (!$default) {
            $default = _t(static::class . '.MISSING_DEFAULT_TITLE', 'No link provided');
        }
        return $default;
```

For all subclasses that just use the description as their default title, please remove those implementations. They can just fall back to using this one instead.

Reply:
Updated